deployment/update_production.sh

```diff
@@ -29,29 +26,21 @@ echo "Starting update..."
 
 set -x # print executed commands. enable this here to not print the if above.
 
-sudo -H -u $USERNAME git fetch
-
-# Note that apache should not be running during most of the upgrade,
-# since then e.g. the backup might be incomplete or the code does not
-# match the database layout, or https://github.com/e-valuation/EvaP/issues/1237.
-[[ -z "$GITHUB_WORKFLOW" ]] && sudo ./deployment/enable_maintenance_mode.sh
```

Comment:
If I recall correctly, we had an in-person discussion where we concluded that it would be fine to just perform this step manually before and after running this script.

The main conflict is that configuring and using apache is external to EvaP itself and it stood in the way of testing the backup process here. I suppose we can keep the commands in the scripts, but I would then have them be opt-in via an environment variable instead of opt-out for CI

Reply:
> but I would then have them be opt-in via an environment variable instead of opt-out for CI

I'd always optimize for the use case, which is the production server here. Forgetting to stop apache can make database operations such as migrations awfully slow, maybe even break, idk. Usages in CI and in a nix dev environment are only for testing. I'd shift all cumbersome-ness to the dev setup.

We can also check if `sudo service --version` runs successfully or some other check and make it depend on that.

Reply:
I just added those lines back now, but I swapped `GITHUB_WORKFLOW` with `EVAP_SKIP_APACHE_STEPS` so that one could run them without apache installed